core/templates/pages/contributor-dashboard-page/contributor-dashboard-page.component.ts

```diff
@@ -50,7 +50,10 @@ export class ContributorDashboardPageComponent
   userCanReviewTranslationSuggestionsInLanguages!: string[];
   userCanReviewVoiceoverSuggestionsInLanguages!: string[];
   userCanReviewQuestions!: boolean;
-  tabsDetails!: ContributorDashboardTabsDetails;
+  tabsDetails!: {
+    [tabName in ContributorDashboardKey]: TabDetails;
+  };
```

Comment:
I don't think you should do this. Instead try this:
At the top of file after the imports
```
const tabs = ContributorDashboardConstants.CONTRIBUTOR_DASHBOARD_TABS_DETAILS;
```
In the component:
```
tabsDetails!: (typeof tabs)[keyof (typeof tabs)];
```

Or you can:
Create utility type: `type valueof<T> = T[keyof T];`

And in the component use:
`tabsDetails!: valueof<typeof ContributorDashboardConstants.CONTRIBUTOR_DASHBOARD_TABS_DETAILS>;

Reply:
@JeeveshGarg Just to check, did you see this comment by @srijanreddy98? It hasn't been responded to.

Reply:
Done

Reply:
This is not done, AFAICT? The old code is still there.

/cc @srijanreddy98 in case the original comment/suggestion needs modification.